Tactility/Source/settings/SystemSettings.cpp

```diff
@@ -40,6 +41,21 @@ static bool loadSystemSettingsFromFile(SystemSettings& properties) {
     bool time_format_24h = time_format_entry == map.end() ? true : (time_format_entry->second == "true");
     properties.timeFormat24h = time_format_24h;
 
+    // Allow system.properties to set the timezone on boot.
+    // Recognized keys:
+    //  - timeZoneCode: the TZ code string
+    //  - timeZoneName: a human name
+    // If timeZoneCode is present we call settings::setTimeZone(name, code).
+    auto tz_code_iter = map.find("timeZoneCode");
+    if (tz_code_iter != map.end()) {
+        std::string tz_name;
+        auto tz_name_iter = map.find("timeZoneName");
+        if (tz_name_iter != map.end()) {
+            tz_name = tz_name_iter->second;
+        }
+        settings::setTimeZone(tz_name, tz_code_iter->second);
+    }
```

Comment:
Any time someone calls `saveSystemSettings()`, this code will break as it will erase the settings that were put there.

Another problem is that the values are not checked for empty string.

Lastly, the neater option would be to only set `timeZoneName`. I'd then use `tt::app::timezone::TimeZoneApp::readTimeZones()` as an example to make a function that iterators over all times zones with a function argument `readTimeZones(std::function<bool(string, string)>)`: the argument would be the read entry and the boolean would determine whether iteration should continue.